crml/erc20-peg/src/lib.rs

```diff
@@ -233,10 +332,111 @@ decl_module! {
 				Erc20Meta::insert(address, (symbol, decimals));
 			}
 		}
+
+		#[weight = 1_000_000]
+		/// Sets the claim delay for a given AssetId
+		pub fn set_claim_delay(origin, asset_id: AssetId, min_balance: Balance, delay: T::BlockNumber) {
+			ensure_root(origin)?;
+			ClaimDelay::<T>::insert(asset_id, (min_balance, delay));
+			Self::deposit_event(<Event<T>>::ClaimDelaySet(asset_id, min_balance, delay));
+		}
 	}
 }
 
 impl<T: Config> Module<T> {
+	/// Process claims at a block after a delay
+	fn process_claim(claim_id: ClaimId) {
+		if let Some(pending_claim) = DelayedClaims::take(claim_id) {
+			match pending_claim {
+				PendingClaim::Deposit((deposit_claim, tx_hash)) => {
+					Self::process_deposit_claim(deposit_claim, tx_hash);
+				}
+				PendingClaim::Withdrawal(withdrawal_message) => {
+					// At this stage it is assumed that a mapping between erc20 to asset id exists for this token
+					let asset_id = Self::erc20_to_asset(withdrawal_message.token_address);
+					if let Some(asset_id) = asset_id {
+						Self::process_withdrawal(withdrawal_message, asset_id);
+					} else {
+						log::error!(
+							"📌 ERC20 withdrawal claim failed unexpectedly: {:?}",
+							withdrawal_message
+						);
+					}
+				}
+			}
+		}
+	}
+
+	fn process_deposit_claim(claim: Erc20DepositEvent, tx_hash: H256) {
+		let event_claim_id = T::EthBridge::submit_event_claim(
+			&Self::contract_address().into(),
+			&T::DepositEventSignature::get().into(),
+			&tx_hash,
+			&EthAbiCodec::encode(&claim),
+		);
+		let beneficiary: T::AccountId = T::AccountId::decode(&mut &claim.beneficiary.0[..]).unwrap();
+		match event_claim_id {
+			Ok(claim_id) => Self::deposit_event(<Event<T>>::Erc20Claim(claim_id, beneficiary)),
```

Comment:
The event deposited earlier had a different claimer id, not beneficiary as second param..
https://github.com/cennznet/cennznet/blob/35fff7139af05f541fefd1167fa3525e1ee5e901/crml/erc20-peg/src/lib.rs#L129